app/controllers/admin_controller.rb

```diff
@@ -2,6 +2,19 @@
 class AdminController < ApplicationController
   layout 'admin/layouts/admin_area'
 
+  before_action :check_admin_ip_whitelist
```

Comment:
This `before_action` config is inherited. The `AdminController` subclasses don't need to do this too. The `before_action :check_admin_ip_whitelist` can be removed from the subclasses `Admin::SettingsController` etc.